src/core/options-container.cpp

```diff
@@ -27,17 +27,47 @@ void options_container::update( std::shared_ptr< extension_snapshot > ext )
     auto ctr = As< options_container >( ext );
     if( ! ctr )
         return;
-    for( auto && opt : ctr->_options )
+    for( auto id : ctr->_ordered_options )
     {
-        _options[opt.first] = opt.second;
+        auto & opt = _options_by_id[id];
```

Comment:
Don't we prefer `at` or any other protected method?
you count or 1 container to have a value on another
If I was Coverity I would shout on that :)

Reply:
Oh it will return a new empty value and you assign it?
Maybe I missed something

Reply:
Right -- if It's new, it will be null and we'll add a new ordered option.